### PR TITLE
Fix Heap Fixup Tracing Schema Failure

### PR DESCRIPTION
**Pull request by jonoommen**
Fix heap fixup tracing schema failures due to previously added heap fixup verbose GC logging and J9mm tracepoints found in https://github.com/eclipse/omr/pull/6593.

Signed-off-by: Jonathan Oommen jon.oommen@gmail.com

### CODE REVIEW

File: gc/verbose/schema.xsd [hunk 1]
```diff
@@ -330,6 +332,9 @@ xmlns:vgc="http://www.ibm.com/j9/verbosegc">
 	</complexType>
 
 	<complexType name="cycle-end">
+		<sequence>
+			<element ref="vgc:heap-fixup" maxOccurs="1" minOccurs="0" />
+		</sequence>
```
Comment (amicic) [inline gc/verbose/schema.xsd:337]: don't understand why is this related to cycle-end!?

Comment (amicic) [reply gc/verbose/schema.xsd:337]: Not expert on this, and from a quick look in xsd files we probably have not way of saying this occurs between cycle-start and cycle-end elements. At least we don't sat it for some other similar tags like allocation-stats, heap-resize etc.

However, we do seem to be able to say it may occur withing gc-op scavenge group....

But that's probably not the same thing, since cycle-start/end are single line independent elements (each have their own closing tag with '/'), while 'gc-ops' are multi-line elements, and can control what subelements end up before their closing tags.

Long story short, this is probably not need, possibly harmful, unless I'm missing something else.

Comment (jonoommen) [reply gc/verbose/schema.xsd]: I see, I will remove this and retest. Could very likely not be needed.

Comment (jonoommen) [reply gc/verbose/schema.xsd]: Retest passed, marking this conversation as resolved.

File: gc/verbose/schema.xsd [hunk 2]
```diff
@@ -624,6 +626,13 @@ xmlns:vgc="http://www.ibm.com/j9/verbosegc">
 		<attribute name="timestamp" type="dateTime" use="optional" />
 	</complexType>
 
+	<complexType name="heap-fixup">
+		<attribute name="id" type="integer" use="optional" />
```
Comment (amicic) [inline gc/verbose/schema.xsd:630]: We don't report id. This is probably a copy-paste from heap-resize which has 2 flavors, and one of them reports ID, hence it's optional here is schema.

ID makes sense to exist if there could be more then one of these elements per cycle. For example, for allocate-stats we don't have it. Very unlikely we could have multiple of them, if possible at all, so I would not bother adding the id in the element, but rather just remove it from the schema.

Comment (jonoommen) [reply gc/verbose/schema.xsd:630]: Ok, yes I can do that.

**Review by amicic — APPROVED** [review]

**Comment by babsingh** [comment]
jenkins build zos

**Comment by babsingh** [comment]
jenkins build zos

**Comment by babsingh** [comment]
jenkins build zos

**Review by babsingh — APPROVED** [review]
